product of unnested array versus not unnested array give very different answers

### What happens?

Two queries that I would expect to give the same answer give different results:

```sql
SELECT product(a) FROM UNNEST(RANGE(1, 11)) _ (a)
```

should produce the same answer as

```sql
SELECT product(a) FROM RANGE(1, 11) _ (a)
```

### To Reproduce

```python
import duckdb

first_sql = "SELECT product(a) FROM UNNEST(RANGE(1, 11)) _ (a)"
second_sql = "SELECT product(a) FROM RANGE(1, 11) _ (a)"

first_answer = duckdb.execute(first_sql).fetchone()[0]
second_answer = duckdb.execute(second_sql).fetchone()[0]

assert first_answer == second_answer, f"{first_answer} != {second_answer}"
```

produces

```
❯ python test.py
Traceback (most recent call last):
  File "/home/cloud/src/github.com/ibis/test.py", line 11, in <module>
    assert first_answer == second_answer, f"{first_answer} != {second_answer}"
AssertionError: 8.173669102896943e+209 != 3628800.0
```

### OS:

Linux

### DuckDB Version:

1.0.1-dev5058

### DuckDB Client:

Python

### Full Name:

Phillip Cloud

### Affiliation:

Voltron Data

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a nightly build

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

--- a/src/execution/operator/scan/physical_table_scan.cpp
+++ b/src/execution/operator/scan/physical_table_scan.cpp
@@ -213,4 +213,13 @@ bool PhysicalTableScan::Equals(const PhysicalOperator &other_p) const {
 	return true;
 }
 
+bool PhysicalTableScan::ParallelSource() const {
+	if (!function.function) {
+		// table in-out functions cannot be executed in parallel as part of a PhysicalTableScan
+		// since they have only a single input row
+		return false;
+	}
+	return true;
+}
+
 } // namespace duckdb

--- a/src/include/duckdb/execution/operator/scan/physical_table_scan.hpp
+++ b/src/include/duckdb/execution/operator/scan/physical_table_scan.hpp
@@ -66,9 +66,7 @@ public:
 	bool IsSource() const override {
 		return true;
 	}
-	bool ParallelSource() const override {
-		return true;
-	}
+	bool ParallelSource() const override;
 
 	bool SupportsBatchIndex() const override {
 		return function.get_batch_index != nullptr;

--- a/test/sql/types/list/unnest_aggregate.test
+++ b/test/sql/types/list/unnest_aggregate.test
@@ -1,0 +1,11 @@
+# name: test/sql/types/list/unnest_aggregate.test
+# description: Test unnest semantics with NULL and empty lists
+# group: [list]
+
+statement ok
+PRAGMA enable_verification
+
+query I
+SELECT SUM(a) FROM UNNEST(RANGE(1, 11)) t(a);
+----
+55

--- a/test/sql/types/list/unnest_aggregate.test
+++ b/test/sql/types/list/unnest_aggregate.test
@@ -9,3 +9,12 @@ query I
 SELECT SUM(a) FROM UNNEST(RANGE(1, 11)) t(a);
 ----
 55
+
+statement ok
+create or replace function rnv(a,b) as (select a + b * pi());
+
+query I
+select rnv(0, 1) from unnest( range(0,2) );
+----
+3.141592653589793
+3.141592653589793

